controllers/upgrade.go

```diff
@@ -140,32 +130,38 @@ func (r *RollingUpgradeContext) ReplaceNodeBatch(batch []*autoscaling.Instance)
 			)
 			//Add statistics
 			r.RollingUpgrade.Status.NodeStep(inProcessingNodes, nodeSteps, r.RollingUpgrade.Spec.AsgName, nodeName, v1alpha1.NodeRotationKickoff)
+		}
+		// Add in-progress tag
+		r.Info("setting batch to in-progress", "instances", awsprovider.GetInstanceIDs(batch), "name", r.RollingUpgrade.NamespacedName())
+		if err := r.Auth.TagEC2instance(batch, instanceStateTagKey, inProgressTagValue); err != nil {
+			r.Error(err, "failed to set batch in-progress", "instances", awsprovider.GetInstanceIDs(batch), "name", r.RollingUpgrade.NamespacedName())
+			return false, err
+		}
 
-			// Add in-progress tag
-			if err := r.Auth.TagEC2instance(instanceID, instanceStateTagKey, inProgressTagValue); err != nil {
-				r.Error(err, "failed to set instance tag", "name", r.RollingUpgrade.NamespacedName(), "instance", instanceID)
-				return false, err
-			}
-
-			// Standby
-			if aws.StringValue(target.LifecycleState) == autoscaling.LifecycleStateInService {
-				r.Info("setting instance to stand-by", "instance", instanceID, "name", r.RollingUpgrade.NamespacedName())
-				if err := r.Auth.SetInstanceStandBy(target, r.RollingUpgrade.Spec.AsgName); err != nil {
-					// failure to set instance to standby are retryable
-					r.Info("failed to set instance to stand-by", "instance", instanceID, "message", err.Error(), "name", r.RollingUpgrade.NamespacedName())
-					return true, nil
-				}
+		// Standby (requeue if any instance was set to stand-by)
+		r.Info("setting batch to stand-by", "instances", awsprovider.GetInstanceIDs(batch), "name", r.RollingUpgrade.NamespacedName())
+		if ok, err := r.Auth.SetInstanceStandBy(batch, r.RollingUpgrade.Spec.AsgName); ok {
```

Comment:
Condition should be `!ok` here

Reply:
I have changed this now. The setTag and setStandby function have been simplified and the rest of the conditions in the controller code.